controllers/windowsmachine_controller.go

```diff
@@ -515,3 +508,21 @@ func (r *WindowsMachineReconciler) isWindowsMachineHealthy(machine *mapi.Machine
 
 	return true
 }
+
+// getInternalIPAddress returns the internal IP address of the Machine
+func getInternalIPAddress(addresses []core.NodeAddress) (string, error) {
+	// Get the IP address associated with the Windows machine, if not error out to requeue again
+	if len(addresses) == 0 {
+		return "", errors.New("no IP addresses defined")
+	}
+	ipAddress := ""
+	for _, address := range addresses {
+		if address.Type == core.NodeInternalIP {
+			ipAddress = address.Address
```

Comment:
Can do a break here to return early once the correct address is found

Reply:
I rewrote the function. PTAL.